docs/api.md

```diff
@@ -248,6 +248,196 @@ after calling this method.
+
+ * If you do the authentication in <a href="#▸-startauthflow">popup</a> mode, you probably want the popup to be closed as soon as the authentication completed to provide a great experience for your users.
+
+<h4 id="auth-init-signature">
+  <a id="auth-init-signature">Signature</a>
+  <a href="#auth-init-signature" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+<code>init(options: <a href="#authinit-options">AuthInitOptions</a>): { status: 'need-auth' | 'authorized' } | <a href="#authfailed-error">AuthFailedError</a></code>
+
+
+<h4 id="authinit-options">
+  <a id="authinit-options"><code>AuthInitOptions</code></a>
+  <a href="#authinit-options" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+```ts
+interface AuthInitOptions {
+  /**
+   * StackBlitz' origin.
+   *
+   * @default https://stackblitz.com
+   */
+  editorOrigin?: string;
+
+  /**
+   * The client id for this OAuth application.
+   */
+  clientId: string;
+
+  /**
+   * OAuth scope.
+   *
+   * @see https://www.rfc-editor.org/rfc/rfc6749#section-3.3
+   */
+  scope: string;
+}
+```
+
+<h4 id="authfailed-error">
+  <a id="authfailed-error"><code>AuthFailedError</code></a>
+  <a href="#authfailed-error" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+```ts
+interface AuthFailedError {
+  status: 'auth-failed';
+
+  /**
+   * A short description of the error.
+   */
+  error: string;
+
+  /**
+   * A detailed description of the error.
+   */
+  description: string;
+}
+```
+
+### ▸ `startAuthFlow`
+
+This starts the OAuth flow, redirecting the current page to the StackBlitz editor to authenticate the user unless `popup` is set to true in which case it's done in a popup.
+
+<code>startAuthFlow(options?: { popup?: boolean }): void</code>
+
+### ▸ `loggedIn`
+
+Returns a promise that resolves when the user authorized your application.
+This promise is guaranteed to never be rejected.
+
+If the user never authorizes or declines your application, this promise never
+resolves.
+
+<h4 id="auth-loggedin-signature">
+  <a id="auth-loggedin-signature">Signature</a>
+  <a href="#auth-loggedin-signature" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+<code>loggedIn(): Promise<void\></code>
+
+<h4 id="auth-loggedin-example">
+  <a id="auth-loggedin-example">Example</a>
+  <a href="#auth-loggedin-example" class="header-anchor" aria-hidden="true">#</a>
+</h4>
+
+```ts
+const instance = await WebContainer.boot();
+
+// wait until the user is logged in
+await auth.loggedIn();
+
+// we can now fetch private packages from our organisation
+await instance.spawn('npm', ['install']);
+```
+
+### ▸ `logout`
+
+Logout the user and clear any credentials that were saved locally.
+
+If `ignoreRevokeError` is set and the revocation failed, the locally-saved credentials are discarded nonetheless.
```

Comment:
Is `revocation` a correct word?

Reply:
I think it is? https://en.wiktionary.org/wiki/revocation#English